pkg/operator/resourcesynccontroller/resourcesync_controller.go

```diff
@@ -0,0 +1,221 @@
+package resourcesynccontroller
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"github.com/golang/glog"
+
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/staticpod/controller/common"
+)
+
+const (
+	operatorStatusResourceSyncControllerFailing = "ResourceSyncControllerFailing"
+	controllerWorkQueueKey                      = "key"
+)
+
+type ResourceSyncController struct {
```

Comment:
go doc. What does this controller do? Please document controllers at least high level. It's very hard for new people to read otherwise.